jsonLoader.go

```diff
@@ -100,25 +104,32 @@ func (l *jsonReferenceLoader) JsonReference() (gojsonreference.JsonReference, er
 	return gojsonreference.NewJsonReference(l.JsonSource().(string))
 }
 
+func (l *jsonReferenceLoader) AddSchema(uri string, b []byte) {
+	l.cachedSchemas[uri] = b
+}
+
 func (l *jsonReferenceLoader) LoaderFactory() JSONLoaderFactory {
 	return &FileSystemJSONLoaderFactory{
-		fs: l.fs,
+		fs:            l.fs,
+		cachedSchemas: l.cachedSchemas,
 	}
 }
 
 // NewReferenceLoader returns a JSON reference loader using the given source and the local OS file system.
 func NewReferenceLoader(source string) *jsonReferenceLoader {
 	return &jsonReferenceLoader{
-		fs:     osFS,
-		source: source,
+		fs:            osFS,
```

Comment:
Now the cache mechanic only works for `jsonReferenceLoader`, but I think it should work for every type of loader, including `StringLoader`, `GoLoader`, etc. Maybe at this point it's worth changing the type to the proper interface as described in #102 and add an `AddSchema` function to the `JSONLoader` interface. The `AddSchema` functionality could be provided by an embedded struct that is embedded in every different loader.